osc_preferences.h

```diff
@@ -0,0 +1,126 @@
+#ifndef __OSC_PREFERENCES_H__
+#define __OSC_PREFERENCES_H__
+
+#include <glib/glist.h>
+#include <glib/gprintf.h>
+
+typedef struct osc_preferences OscPreferences;
+typedef struct oscplot_preferences OscPlotPreferences;
+typedef struct iio_device_preferences IioDevPreferences;
+typedef struct iio_channel_preferences IioChnPreferences;
+
+struct oscplot_preferences {
+        unsigned int *sample_count;
+        GList *devices_pref_list;
+};
+
+struct iio_channel_preferences {
+        char *chn_name;
+        unsigned int *constraints;
+};
+
+struct iio_device_preferences {
+        char *dev_name;
+        double *fft_correction;
+        GList *channels_pref_list;
+};
+
+struct osc_preferences {
+        struct oscplot_preferences *plot_preferences;
+};
+
+
+/* Construct/Destruct methods */
+
+
+/* IioChnPreferences */
+
+static inline IioChnPreferences* iio_chn_preferences_new(const char *chn_name)
+{
+        IioChnPreferences *chn_pref = g_new(IioChnPreferences, 1);
+
+        chn_pref->chn_name = g_strdup(chn_name);
+        chn_pref->constraints = NULL;
+
+        return chn_pref;
+}
+
+static inline void iio_chn_preferences_free(IioChnPreferences *pref)
```

Comment:
just as a note: these `inline` specifiers don't do anything for these free callback/functions, since these functions are also passed via function pointers to some Glib list cleanup thingi;
inline usually works when the compiler can expand this code as a "macro"; and based on some internal logic it does or doesn't do that

they are fine to leave here though

Reply:
I used the inline just to avoid duplicate symbols since this is included in multiple places. I will move the functions implementation to a .c file in order to hide implementation details.